crates/fulf/src/ascii/mod.rs

```diff
@@ -0,0 +1,159 @@
+//! Working with ASCII-only strings.
+//!
+//! Cheating!
+
+mod bytelines;
+pub use bytelines::ByteLines;
+mod matcher;
+pub use matcher::{ascii_from_bytes, matcher};
+
+use crate::scoring_utils::*;
+
+#[inline]
+pub fn score_with_positions(needle: &[u8], haystack: &[u8]) -> (Score, Vec<usize>) {
```

Comment:
hmm, why we are duplicating this here?

Reply:
> hmm, why we are duplicating this here?

It was easier to change the code in just one crate, plus, `ascii` module is a bit more than just fzy algorithm, if you are asking why I duplicate this with "extracted_fzy". Of course, those two crates should not duplicate the code, but I'm not sure how they should be merged. Probably, algorithm-specific things should be moved out of the `fulf`.

Or, if you are asking why there is "utf8" and "ascii" with duplicated code - that's because they work with different types: ascii algorithm works with byteslices and bytes, where utf8 algorithm works with string and chars.

Reply:
To be merged into the master, we better at least refactor the duplicated `extracted_fzy` code in case of the bug of `extracted_fzy` like https://github.com/liuchengxu/vim-clap/pull/409/files#diff-744145d5678fb6819a7b4fddd9b56248R94 once we finish the other works in this PR.

Reply:
> To be merged into the master, we better at least refactor the duplicated `extracted_fzy` code in case of the bug of `extracted_fzy` like https://github.com/liuchengxu/vim-clap/pull/409/files#diff-744145d5678fb6819a7b4fddd9b56248R94 once we finish the other works in this PR.

I copied the fixed algorithm, and added the ASCII there, watch c7a7a94 commit. Should be fine.